source/agora/consensus/data/Transaction.d

```diff
@@ -48,6 +49,61 @@ public enum TxType : ubyte
 
 public struct Transaction
 {
+    @safe:
+
+    /// ctor with only outputs
+    this (in TxType type, in Output[] outputs) nothrow
+    {
+        this.type = type;
+        this.outputs = outputs.dup; // .sort.array; (FIXME: fix compilation error)
+    }
```

Comment:
Accept a mutable value (let the caller `dup` if needed) and just call `sort` ?

Reply:
Updated